### ISSUE TITLE
can you use pixel streaming 5.1 branch with unreal engine 5.2?

### ISSUE THREAD

**Issue by comsyspro** [EpicGames/PixelStreamingInfrastructure-archived-2024#219]
i've seen now that pixel streaming has developed further since ue 5.0. now i'm asking me if you can use use pixel streaming 5.1 branch with unreal engine 5.2? is pixel streaming 5.2 a dev branch?

also when looking into the code on branch 5.1 there is the "app.js" available (https://github.com/EpicGames/PixelStreamingInfrastructure/tree/UE5.1/SignallingWebServer/scripts - like in the 5.0 version) whereas in 5.2 branch i can't see any app.js or scripts folder. do you have to compile the javascript files by yourself or how to get app.js in 5.2 branch?

so what version of pixel streaming you should use for a production case in combination with unreal engine 5.2 projects?

**Comment by lukehb** [comment]
> whereas in 5.2 branch i can't see any app.js or scripts folder. do you have to compile the javascript files by yourself or how to get app.js in 5.2 branch?

In 5.2 we have completely changed from a monolothic javascript files to using a library. In short app.js does not exist anymore and is replaced with a library that we use to build the so-called "reference frontend", which looks and feels a lot like the old app.js. 

Your options:

- Compile the typescript library yourself and build the reference frontend ( https://github.com/EpicGames/PixelStreamingInfrastructure/tree/eea7839f513056ebbd0f98930eb8b8ca895d6bf1/Frontend/implementations/EpicGames )
- Download our releases ( https://github.com/EpicGames/PixelStreamingInfrastructure/releases ) which contain the built reference frontend for your convenience  ( specifically `player.html` and `player.js` ).

> so what version of pixel streaming you should use for a production case in combination with unreal engine 5.2 projects?

Use the UE5.2 branch of this repo or the associated releases marked with UE5.2. Additionally, when you build projects using Unreal Engine 5.2 and have Pixel Streaming enabled you will see in your packaged output you end up with a `Samples/PixelStreaming` folder, that directory has a script in it called `get_ps_servers.bat` you can run and it will download and extract the latest relevant release from this repository for you.

> can you use pixel streaming 5.1 branch with unreal engine 5.2?

In short, no. Use the 5.2 branch for 5.2 projects.
